src/components/TransactionPreview/HistoryPreview.tsx

```diff
@@ -66,32 +66,25 @@ const HistoryPreview = ({ crossChainAction }: TransactionPreviewInterface) => {
     // Convert gas wei to ETH
     const gasInETHAmount = gasCost / 1e18;
 
-    // Convert to USD
-    let gasFees = gasInETHAmount * crossChainAction.estimated.usdPrice;
-    let gasFeesNative = gasInETHAmount;
+    // calculate gas fees
+    const gasfees = gasInETHAmount * asset.gasUsed;
 
-    gasFeesNative = parseFloat(gasInETHAmount.toFixed(4)) < 0.0001 ? '<0.0001' : gasInETHAmount.toFixed(5);
-    const partialgasFees = gasFeesNative.toString();
-    let fullGasFees = gasInETHAmount.toFixed(15);
-    const finalgasFees = fullGasFees.toString();
-    console.log(fullGasFees);
+    // convert gas fee to USD
+    const gasfeesInUsd = gasfees * crossChainAction.estimated.usdPrice;
+    const fullgasFee = gasfeesInUsd.toFixed(6);
+    const partialgasfeesInUsd = parseFloat(gasfeesInUsd.toFixed(3)) < 0.001 ? '<0.001' : gasfeesInUsd.toFixed(2);
```

Comment:
please follow camelCase naming standards

```suggestion
    const partialGasFeesInUsd = parseFloat(gasfeesInUsd.toFixed(3)) < 0.001 ? '<0.001' : gasfeesInUsd.toFixed(2);
```